core/common/src/geometry/Cartographic.ts

```diff
@@ -6,45 +6,68 @@
+/** JSON representation of a Cartographic object.
+ * @see [[Cartographic]]
+ * @public
+ * */
+export interface CartographicProps {
+  /** Angle which represents latitude. */
+  latitude: AngleProps;
+  /** Angle which represents longitude. */
+  longitude: AngleProps;
+  /** Height in meters above the ellipsoid; defaults to zero. */
+  height?: number;
+}
 
 /** A position on the earth defined by longitude, latitude, and height above the [WGS84](https://en.wikipedia.org/wiki/World_Geodetic_System) ellipsoid.
  * @public
  */
-export class Cartographic implements LatLongAndHeight {
+export class Cartographic {
   /**
-   * @param longitude longitude, in radians.
-   * @param latitude latitude, in radians.
+   * @param longitude longitude, as an Angle object.
+   * @param latitude latitude, as an Angle object.
    * @param height The height, in meters, above the ellipsoid.
    */
-  constructor(public longitude: number = 0, public latitude: number = 0, public height: number = 0) { }
+  private constructor(public longitude: Angle, public latitude: Angle, public height: number = 0) { }
 
-  /** Create a new Cartographic from longitude and latitude specified in radians.
-   * @param longitude longitude, in radians.
-   * @param latitude latitude, in radians.
-   * @param height The height, in meters, above the ellipsoid.
-   * @param result The object onto which to store the result.
+  /** Create an empty Cartographic object */
+  public static createEmpty(): Cartographic {
+    return Cartographic.fromJSON({longitude: 0, latitude: 0, height: 0});
+  }
+
+  /** Creates a Cartographic object from a JSON representation.
+   * @param props object containing longitude, latitude, and height information.
+   * @see [[CartographicProps]]
    */
-  public static fromRadians(longitude: number, latitude: number, height: number = 0, result?: Cartographic) {
+  public static fromJSON(props: CartographicProps, result?: Cartographic): Cartographic {
+    // let longitude = 0;
+    // let latitude = 0;
+
+    // if (true === props.isDegrees) { // ensure stored in radians
+    //   longitude = Angle.degreesToRadians(props.longitude);
+    //   latitude = Angle.degreesToRadians(props.latitude);
+    // } else { // already stored in radians
+    //   longitude = props.longitude;
+    //   latitude = props.latitude;
+    // }
+    const height = props.height !== undefined ? props.height : 0;
+
     if (!result)
-      return new Cartographic(longitude, latitude, height);
+      return new Cartographic(Angle.fromJSON(props.longitude), Angle.fromJSON(props.latitude), height);
 
-    result.longitude = longitude;
-    result.latitude = latitude;
+    result.longitude = Angle.fromJSON(props.longitude);
+    result.latitude = Angle.fromJSON(props.latitude);
     result.height = height;
     return result;
   }
 
-  public toJSON(): LatLongAndHeight {
+  /** Returns a JSON representation of a Cartographic object. */
+  public toJSON(): CartographicProps {
     return {
```

Comment:
Shallow copy = bad.

Reply:
just changed that; thanks.  changed a few other places but neglected this place.